src/course-outline/CourseOutline.test.jsx

```diff
@@ -518,46 +519,105 @@ describe('<CourseOutline />', () => {
     });
   });
 
-  it('check section list is ordered successfully', async () => {
-    const { getAllByTestId } = render(<RootWrapper />);
+  it('check that new section list is saved when dragged', async () => {
+    const { getAllByRole } = render(<RootWrapper />);
+
     const courseBlockId = courseOutlineIndexMock.courseStructure.id;
-    let { children } = courseOutlineIndexMock.courseStructure.childInfo;
-    children = children.splice(2, 0, children.splice(0, 1)[0]);
+    await waitFor(async () => {
+      const sectionsDraggers = await getAllByRole('button', { name: 'Drag to reorder' });
 
-    axiosMock
-      .onPut(getEnableHighlightsEmailsApiUrl(courseBlockId), { children })
-      .reply(200, { dummy: 'value' });
+      axiosMock
+        .onPut(getCourseBlockApiUrl(courseBlockId))
+        .reply(200, { dummy: 'value' });
 
-    await executeThunk(setSectionOrderListQuery(courseBlockId, children, () => {}), store.dispatch);
+      const section1 = store.getState().courseOutline.sectionsList[0].id;
+      const draggableButton = sectionsDraggers[7];
+      await act(async () => fireEvent.keyDown(draggableButton, { code: 'Space' }));
+      await act(async () => fireEvent.keyDown(draggableButton, { key: 'ArrowUp' }));
+      await act(async () => fireEvent.keyDown(draggableButton, { code: 'Space' }));
```

Comment:
You shouldn't need to use act here since fireEvent should already be using it internally.

Reply:
Have removed most of the fireEvents. Removign act() from the space event doesn't fire the UI update. Seems to be a `dnd-kit` quirk, the other way is [to not wrap in act, but do a sleep(1) after the space keypresses](https://github.com/clauderic/dnd-kit/issues/261). act() seemed more react friendly